app/design/adminhtml/default/default/template/widget/grid/massaction.phtml

```diff
@@ -48,11 +48,23 @@
                     <?php echo $this->getBlockHtml('formkey')?>
                     <fieldset>
                         <span class="field-row">
-                            <label><?php echo $this->__('Actions') ?></label>
+                            <label for="<?php echo $this->getHtmlId() ?>-select"><?php echo $this->__('Actions') ?></label>
                             <select id="<?php echo $this->getHtmlId() ?>-select" class="required-entry select absolute-advice local-validation">
                                 <option value=""></option>
-                                <?php foreach($this->getItems() as $_item): ?>
-                                    <option value="<?php echo $_item->getId() ?>"<?php echo ($_item->getSelected() ? ' selected="selected"' : '')?>><?php echo $_item->getLabel() ?></option>
+                                <?php foreach($this->getGroupedItems() as $key => $group): ?>
+                                    <?php if ($key === 'default'): ?>
+                                        <?php foreach($group as $_item): ?>
+                                            <option value="<?php echo $_item->getId() ?>"<?php echo ($_item->getSelected() ? ' selected="selected"' : '')?>><?php echo $_item->getLabel() ?></option>
+                                        <?php endforeach ?>
+                                    <?php elseif ($key === 'grouped'): ?>
+                                        <?php foreach($group as $label => $_massGroup): ?>
+                                            <optgroup label="<?php echo $this->quoteEscape($label); ?>">
+                                                <?php foreach($_massGroup as $_item): ?>
+                                                    <option value="<?php echo $_item->getId() ?>"<?php echo ($_item->getSelected() ? ' selected="selected"' : '')?>><?php echo $_item->getLabel() ?></option>
+                                                <?php endforeach ?>
+                                            </optgroup>
+                                        <?php endforeach ?>
+                                    <?php endif ?>
```

Comment:
```suggestion
                                <?php foreach($this->getGroupedItems() as $_group => $_items): ?>
                                    <?php if ($_group !== 'default'): ?>
                                        <optgroup label="<?php echo $this->quoteEscape($_group); ?>">
                                    <?php endif; ?>
                                    <?php foreach($_items as $_item): ?>
                                        <option value="<?php echo $_item->getId() ?>"<?php echo ($_item->getSelected() ? ' selected="selected"' : '')?>><?php echo $_item->getLabel() ?></option>
                                    <?php endforeach ?>
                                    <?php if ($_group !== 'default'): ?>
                                        </optgroup>
                                    <?php endif; ?>
```

Withou the extra grouped key https://github.com/OpenMage/magento-lts/pull/1913/files#r773062078

It would look like this.

Reply:
It is duscutable....In my solution someone can get all grouped keys just calling of `$this->getGroupedItems()['grouped']`. In your solution you have to iterate all items and put to new array for this purpose.

Reply:
I am not saying it's wrong :) I just like the `default` group on the same level as the other groups.
BTW to get all grouped options you would just unset `default`

Reply:
🚔 
You have forgotten a space for `foreach(` + `)?>` and there are no ; for `<?php endif; ?>`.
You can also remove the unnecessary parentheses for `echo ($_item->getSelected() ? ' selected="selected"' : '')`.
🤐